source/rules/no-finnish.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { typecheck } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows the use of Finnish notation.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Finnish notation is forbidden"
+    },
+    schema: []
+  },
+  create: context => {
+    const { couldBeObservable } = typecheck(context);
+
+    function report(
+      node: es.Node | es.FunctionDeclaration | es.FunctionExpression
+    ) {
+      if (couldBeObservable(node)) {
+        let reportObject = { messageId: "forbidden", node };
+
+        if ((node as es.FunctionDeclaration).id) {
+          const funcId = (node as es.FunctionDeclaration).id;
+          reportObject = Object.assign({}, reportObject, { node: funcId });
+        }
+        if ((node as es.MethodDefinition).key) {
+          const funcId = (node as es.MethodDefinition).key;
+          reportObject = Object.assign({}, reportObject, { node: funcId });
+        }
+
+        context.report(reportObject);
+      }
+    }
+
+    function isErrorWorthy(node: any): boolean {
+      let name;
+      if (node.key) {
+        name = node.key.name;
+      } else {
+        name = node.name;
+      }
+
+      return /\$$/.test(name) && couldBeObservable(node);
+    }
+
+    function reportInterface(node: any) {
```

Comment:
I feel that something can be done with this function/test.
and also, if anyone can suggest on how to keep inference. I cannot get the inference to work (i.e. I have to use any  😫)